src/base/math/range.h

```diff
@@ -17,6 +17,14 @@ template <typename T> struct Range
 		double shift;
 	};
 
+	static Range<T> Raw(const T &min, const T &max)
```

Comment:
Is the 2 parameter constructor used anywhere? If not, it can be used as not min-max construction. If used I think the min-max construction should be a factory function and not the raw

Reply:
It is used. Please provide an explanation for your opinion why do you think the current constructor should be the factory.

Reply:
I think this range class can be aggregate type. (aggregate: no user-defined constructor exists, and all class members should be public.) That will make `consteval static auto members()` function unnecessary.

I don't want you to make this type of aggregate, but the long-term goal is this I think.

Reply:
This is an exception use-case where we need an irregular range. All other uses are needed to be "normalized"

Reply:
Another problem is that the 2-parameter constructor assumes an invariant that min should be lower than max. This invariant is not true, and probably that is why it created this factory function

Reply:
OK, then no changes required

Reply:
What about something like this?
```c++
template <typename T, bool positive = true> struct Range
{
  	template <typename = std::enable_if_t<positive>>
	Range(const T &x, const T &y) :
	    min(std::min(x, y)),
	    max(std::max(x, y))
	{}
```

Reply:
i agree. this PR is a quick fix on the release candidate.